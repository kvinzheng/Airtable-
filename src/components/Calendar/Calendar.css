.Calendar {
    margin: 0 auto;
    width: 90%;
    min-width: 60em;
}