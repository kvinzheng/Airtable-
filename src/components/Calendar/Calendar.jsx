import React, { useEffect, useState, useRef } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import CalendarContent from "../CalendarContent/CalendarContent";
import CalendarHeader from "../CalendarHeader/CalendarHeader";
import CalendarForm from "../../common/Form/CalendarForm";

import { setEvents } from "../../actions/events";
import { calculatePosition } from "../../helpers/eventTimeline";

import "./Calendar.css";

export const Calendar = ({ events, form }) => {
  const [displayedDate, setDisplayedDate] = useState(new Date("2018/01/01"));
  const [eventPosition, setEventPosition] = useState([]);

  const calendarRef = useRef();

  useEffect(() => {
    const position = [];
    events.forEach((item) => {
      calculatePosition([item.id, item.date, item.end], position);
      setEventPosition(position);
    });
  }, [events]);

  const handleMonthShift = (date) => {
    setDisplayedDate(date);
  };

  return (
    <div className="Calendar" data-testid="Calendar" ref={calendarRef}>
      {form && (
        <CalendarForm
          data-testid="CalendarForm"
          displayedDate={displayedDate}
        />
      )}
      <CalendarHeader
        data-testid="CalendarHeader"
        displayedDate={displayedDate}
        onMonthShift={handleMonthShift}
      />
      <CalendarContent
        data-testid="CalendarContent"
        displayedDate={displayedDate}
        outlinedDate={new Date()}
        eventPosition={eventPosition}
      />
    </div>
  );
};

export const mapStateToProps = (state) => {
  return {
    events: state.events.data,
    form: state.form.calendarForm,
  };
};

Calendar.propTypes = {
  events: PropTypes.array,
  form: PropTypes.bool,
};

export default connect(mapStateToProps, { setEvents })(Calendar);
