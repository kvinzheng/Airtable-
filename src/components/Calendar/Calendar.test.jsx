import React from "react";
import thunk from "redux-thunk";

import configureMockStore from "redux-mock-store";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { Calendar } from "./Calendar";
import { mapStateToProps } from "./Calendar";

import MockData from "../../helpers/testingData";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

configure({ adapter: new Adapter() });

describe("Calendar Component", () => {
  it("should render without any data defined", () => {
    const component = shallow(<Calendar />);

    expect(component).toMatchSnapshot();
  });

  it("should render with props passed in", () => {
    const component = shallow(
      <Calendar
        events={MockData.events.data}
        form={MockData.form.calendarForm}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("mapStateToProps", () => {
    const state = {
      events: { data: [] },
      form: { calendarForm: false },
    };
    const expected = {
      events: [],
      form: false,
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });
});
