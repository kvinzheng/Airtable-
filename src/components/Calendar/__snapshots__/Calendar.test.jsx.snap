// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Component should render with props passed in 1`] = `ShallowWrapper {}`;

exports[`Calendar Component should render without any data defined 1`] = `ShallowWrapper {}`;
